specs/pvs/corla.pvs

```diff
@@ -2022,7 +2066,111 @@ BEGIN
 
-%+BEGIN data_model
+%+END data_model
+
+%+BEGIN protocol
+
+%%% \section{Protocol}
+%%% \label{sec:Protocol}
+%%%
+%%% As discussed
+%%% in~\Cref{sec:Background_Model_Elements,sec:System_Architecture},
+%%% the RLA Tool is a client-server system. In this section we
+%%% describe the protocol used by the client and server to communicate
+%%% with each other.
+
+protocol: THEORY
+BEGIN
+  IMPORTING system_architecture_component, system_architecture
+
+%%% The client-side of the system is exclusively a web browser running
+%%% an HTML/Javascript-based web application.  Recall that
+%%% in~\Cref{sec:System_Architecture} we mandated that all client
+%%% artifacts (static content, HTML, and Javascript) are provided
+%%% directly from the (web) server.
+
+  client: VAR front_end
+  browser: VAR web_browser
+
+%%% The server-side of the system is a web server which communicates
+%%% over HTTPS. It serves both static content, such as images and HTML
+%%% fragments, and executable content, such as Javascript code. As
+%%% such, this is a standard REST-based, multi-tier architecture.
+%%%
+%%% To provide assurance about system availability, we deploy multiple
+%%% instances of web and application servers providing content and
+%%% databases providing persistent data storage. All of these
+%%% artifacts are connected over multiple networks in CDOS's hosting
+%%% service. We underspecify the details of the deployment network
+%%% configuration here at the request of CDOS.
+
+  server: VAR back_end
+  servers: VAR set[server]
+  web_server: VAR web_server
+  application_server: VAR application_server
+  networks: VAR set[network]
+  databases: VAR set[database]
+
+%%% Multiple databases run in a failover configuration. Web
+%%% application servers communicate with databases over a standard
+%%% secure remote SQL connection. Web servers serve static content and
+%%% delegate dynamic content services to the application servers.
+%%%
+%%% All communication between the client web browsers and the
+%%% server(s) is initiated by the client, as with normal REST-based
+%%% architectures. As such, functions defined in the dashboards
+%%% (\Cref{sec:Dashboards}) of the system are the endpoint of all
+%%% communication in the system. The RLA Tool protocol is consequently
+%%% the union of all of those endpoints.
+%%%
+%%% \subsection{State-wide Dashboard Protocol}
+%%% \label{sec:State-wide_Dashboard_Protocol}
+%%%
+%%% As discussed in~\Cref{sec:Department_of_State_Dashboard}, the
+%%% Department of State Dashboard has eight (8) features: one for
+%%% authentication, six for providing information about a given
+%%% election and RLA, and one for getting updates on the current RLA.
+%%%
+
+%%% \begin{enumerate}
+
+%%% \item \texttt{authenticate\_state\_administrator}: authentication
+%%% of state administrators (the mandatory first step in the
+%%% protocol),
+
+%%% \item \texttt{select\_contest\_for\_comparison\_audit}: defining
+%%% the elections under audit,
```

Comment:
It is supposed to be "selecting contest" in the comment, of course.  As contests relate to elections, there is no need to pass an election.  The refresh endpoint provides the lists you mention, @nealmcb.  

At the state level CDOS has asked for a list of ballots to be audited in each contest (possibly spanning counties).  At the county dashboard level, they'll see the (filtered) list of ballots for their county.